sim/warlock/pets.go

```diff
@@ -62,6 +68,16 @@ func (warlock *Warlock) makePet(summonType proto.WarlockOptions_Summon, baseStat
 	if attackOptions != nil {
 		pet.EnableAutoAttacks(pet, *attackOptions)
 	}
+	if warlock.Spec == proto.Spec_SpecDemonologyWarlock {
+		masteryBonus := func(mastery float64) float64 {
+			return 1 + math.Floor(2.3*core.MasteryRatingToMasteryPoints(mastery))/100.0
+		}
+
+		pet.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery float64, newMastery float64) {
+			pet.PseudoStats.DamageDealtMultiplier /= masteryBonus(oldMastery)
+			pet.PseudoStats.DamageDealtMultiplier *= masteryBonus(newMastery)
+		})
+	}
```

Comment:
This should be done in the pets Initialize method. Also right now you're removing the oldMastery bonus from the pet but I dont see it ever being activated before this callback. What will happen is that you are probably removing a non applied multiplier.

Reply:
I guess moving it into initialize is cleaner, but should be functionally identical right?

I'm indeed removing the old mastery bonus, but since mastery at the start is 0 that multiplier is 1.0, so it doesn't do anything.

Reply:
This callback is called only for dynamic stat changes during runs. If you apply the DamageMulti in the Initialize of the pet it should match the value of oldMastery in the first callback

Reply:
Pets can't possibly have mastery at the start, so I don't need anything in initialize. The only way they can get it is via stat inheritance and that is dynamically applied on summon.